### PR TITLE
Fix/support concurrent devices system test

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
#### new system test code

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- new implementation for the cbl system test

### CODE REVIEW

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    db_name = cbl_params["db_name"]
+    db_obj = cbl_params["db_obj"]
+    platform = cbl_params["platform"]
+
+    thread_name = "Thread {}".format(db_name)
+
+    try:
+        repl_obj = Replication(base_url)
+        authenticator = Authenticator(base_url)
+        cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username, ttl=900000)
```
Comment (mgcouchbase) [inline testsuites/CBLTester/system_tests/test_system_concurrent_testing.py:202]: Can we define the ttl in separate variable , for readability

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+from libraries.data.doc_generators import simple, four_k, simple_user, complex_doc
+from datetime import datetime, timedelta
+from CBLClient.Utils import Utils
+
+
+@pytest.mark.listener
+@pytest.mark.replication
+def test_system(params_from_base_suite_setup):
+    sync_gateway_version = params_from_base_suite_setup["sync_gateway_version"]
+    if sync_gateway_version < "2.0.0":
```
Comment (mgcouchbase) [inline testsuites/CBLTester/system_tests/test_system_concurrent_testing.py:21]: Like Functional test summary , can you please write the steps? Also match it with the comments. For future quick debug.

**Review by mgcouchbase — COMMENTED** [review]
Nice work Eunice

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+            current_time = datetime.now()
+
+        # stopping replication
+        _log_system_test("Thread {}".format(db_name), "test finishes", "Test completed. Stopping Replicators")
+
+        repl_obj.stop(repl)
+        time.sleep(5)
+        _check_doc_count(db_obj, cbl_db)
+
+        results = query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
```
Comment (sridevi-15) [inline testsuites/CBLTester/system_tests/test_system_concurrent_testing.py:354]: looks like you collected query results, but not using in the test after this. Can you make sure you verify the query results?

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+
+        return {db_name: doc_ids}
+    except (Exception, RuntimeError) as ex:
+        _log_system_test(thread_name, "exception caught", ex)
+        _log_system_test(thread_name, "quit smoothly")
+    finally:
```
Comment (sridevi-15) [inline testsuites/CBLTester/system_tests/test_system_concurrent_testing.py:360]: We do not need to have as part of this PR. Can we do logs scanning at the end of the system test to check all CBLs and SGW. For SGW we can integrate with eagle eye, but for atleast do logs scanning for all cbls. We can have it as enhancement as next of this

**Review by sridevi-15 — CHANGES_REQUESTED** [review]

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py (hunk 1 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/system_tests/test_system_concurrent_testing.py]: done

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py (hunk 2 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/system_tests/test_system_concurrent_testing.py:35]: added summary and steps

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py (hunk 3 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/system_tests/test_system_concurrent_testing.py]: fixed

File: testsuites/CBLTester/system_tests/test_system_concurrent_testing.py (hunk 4 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/system_tests/test_system_concurrent_testing.py:376]: have logged a ticket, will be discussing it with server qe team.

**Review by sridevi-15 — APPROVED** [review]

**Review by mgcouchbase — APPROVED** [review]

**Pull request merged**